src/aoutils.cpp

```diff
@@ -0,0 +1,81 @@
+#include "aoutils.h"
+
+#include <QDebug>
+#include <QMap>
+#include <QPair>
+#include <QRegularExpression>
+#include <QRegularExpressionMatch>
+#include <QSettings>
+
+void AOUtils::migrateEffects(QSettings &p_effects_ini)
+{
+  qDebug() << "Migrating effects from file:" << p_effects_ini.fileName();
+
+  QMap<QString, QString> l_effect_map;
+  {
+    const QStringList l_key_list = p_effects_ini.childKeys();
+    for (const QString &i_key : l_key_list)
+    {
+      l_effect_map.insert(i_key, p_effects_ini.value(i_key).toString());
+    }
+
+    p_effects_ini.clear();
+    p_effects_ini.sync();
+  }
```

Comment:
why is this in a block

Reply:
Mostly to highlight it visually since its a destructive action on the ini as it clears the entire file before adding the info back in the new format.